legit.py

```diff
@@ -0,0 +1,230 @@
+import logging
+from typing import Iterable
+
+from git import Repo
+from git.exc import GitCommandError
+
+log = logging.getLogger("legit")
+log.setLevel(logging.DEBUG)
+fh = logging.FileHandler("legit.log")
+fh.setLevel(logging.DEBUG)
+ch = logging.StreamHandler()
+ch.setLevel(logging.ERROR)
+formatter = logging.Formatter("%(asctime)s\t%(levelname)s -- %(processName)s %(filename)s:%(lineno)s -- %(message)s")
+ch.setFormatter(formatter)
+fh.setFormatter(formatter)
+log.addHandler(ch)
+log.addHandler(fh)
+
+
+class Repository(object):
+    """GitPython provides object model access to your git repository.
+    This class wraps up some fuctions of GitPython
+
+    Attributes:
+        repo : This is path to git repository. Most of the GitPython functions
+        use it to operate with repository.
+
+    """
+
+    def __init__(self, path: str) -> None:
+        """This command creates an empty Git repository - basically
+        a .git directory with subdirectories for objects,
+        refs/heads, refs/tags, and template files.
+        An initial HEAD file that references the HEAD
+        of the master branch is also created.
+
+        Note:
+            Git accordance is "git init `path`"
+            https://git-scm.com/docs/git-init
+
+        Args:
+            path: Path to the repository you want to be git repository
+
+        """
+        log.debug(f"git init {path}")
+        try:
+            self.repo = Repo.init(path)
+        except GitCommandError:
+            logging.exception(f"GitCommandError: git init {GitCommandError}, {path} traceback")
+
+    @classmethod
+    def clone(cls, url: str, path: str = None) -> str:
+        """Git function which provides interface to clone from remote
+        to local machine.
+
+        Note:
+            Git accordance is "git clone `url` `path`"
+            https://git-scm.com/docs/git-clone
+
+        Args:
+            url: The url of remote (Github, Gitlab, etc)
```

Comment:
of remote, fullstop. If the user needs guidance on git, s/he can look up the git manual. Here we are documenting the method to work with git, we can assume that the user knows what git is if  s/he has picked up this module